docs/networking/index.md

```diff
@@ -13,6 +13,37 @@ user <-> workspace connections are end-to-end encrypted.
 
 [Tailscale's open source](https://tailscale.com) backs our networking logic.
 
+## Requirements
+
+In order for clients and workspaces to be able to connect:
+
+- All clients and agents must be able to establish a connection to the Coder
+  server (`CODER_ACCESS_URL`) over HTTPS (`tcp/443`).
+- Any reverse proxy or ingress between the Coder control plane and clients must
+  support WebSockets.
+
+In order for clients to be able to establish direct connections:
+
+> **Note:** Direct connections via the web browser are not supported. To improve
+> latency for browser-based applications running inside Coder workspaces,
+> consider deploying one or more
+> [workspace proxies](../admin/workspace-proxies.md).
+
+- The client is connecting using the CLI (e.g. `coder ssh` or
+  `coder port-forward`).
+- The client and workspace agent are both able to connect to a specific STUN
+  server.
+  > The STUN server needs to tell the client and workspace their respective
+  > `address:port` pairs from its perspective so that they can establish a
+  > direct connection with each other. If the client and agent are only able to
+  > connect to STUN servers on different networks, then a direct connection will
+  > not be possible. For an in-depth technical explanation, see
+  > [How NAT traversal works (tailscale.com)](https://tailscale.com/blog/how-nat-traversal-works).
+- Outbound UDP traffic must be allowed for both the client and the agent from
+  source ports `udp/3478` and `udp/41641` to all destination ports.
+  > For more detailed information, see
+  > [What firewall ports should I open to use Tailscale? (tailscale.com)](https://tailscale.com/kb/1082/firewall-ports).
+
```

Comment:
There should probably be some details about NATs and stuff but I really don't know what to write without getting super technical :/

Reply:
Honestly I'd prefer to just link to Tailscale's docs on this.

Reply:
IDK how well Tailscale's docs cut it, this one doesn't really explain much about NAT just about firewalls.

Reply:
I link to https://tailscale.com/blog/how-nat-traversal-works a bit above in the STUN section.

Reply:
This is a good blog post but it's not very good documentation since it's 9000 words long. We should probably dumb it down

Reply:
OK, I added a fairly high-level overview.

Reply:
I've Slack'd you some diagrams.  If we're going to explain it, then those are my suggested examples.

Reply:
Added and moved to a separate page as this one is getting plenty big already.